build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/DependencyLicensesTask.java

```diff
@@ -114,6 +114,11 @@
      */
     private Set<String> ignoreShas = new HashSet<>();
 
+    /**
+     *  Names of files that should be ignored by the check
+     */
+    private Set<String> ignoreFiles = new HashSet<>();
```

Comment:
Can you add a getIgnoreFiles() with the following annotations. 
```java
    @Input
    @Optional
```
This helps Gradle decide if the input has changed and needs to be re-run (i.e. if I change configuration in build.gradle for the ingoredFiles, then this gradle task needs to be re-run): https://docs.gradle.org/current/userguide/custom_tasks.html#sec:which_inputs_are_considered_out_of_date

Reply:
Yes we need to track these ignoredFiles as inputs. Otherwise we break the up-to-date check for those tasks

Reply:
Added